src/tests/Microsoft.PowerFx.Core.Tests/LexerTests.cs

```diff
@@ -568,5 +568,71 @@ public void TestUnsupportedDecimalSeparatorCausesFallback()
 
             CultureInfo.CurrentCulture = oldCulture;
         }
+
+        [Fact]
+        public void TestStringInterpolation()
+        {
+            Token[] tokens;
+            string value;
+
+            value = "$\"Hello {name}\"";
+            tokens = TexlLexer.LocalizedInstance.LexSource(value);
+            Assert.NotNull(tokens);
+            Assert.Equal(7, tokens.Length);
+            Assert.Equal(TokKind.StrInterpStart, tokens[0].Kind);
+            Assert.Equal(TokKind.StrLit, tokens[1].Kind);
+            Assert.Equal(TokKind.IslandStart, tokens[2].Kind);
+            Assert.Equal(TokKind.Ident, tokens[3].Kind);
+            Assert.Equal(TokKind.IslandEnd, tokens[4].Kind);
+            Assert.Equal(TokKind.StrInterpEnd, tokens[5].Kind);
+            Assert.Equal(TokKind.Eof, tokens[6].Kind);
+
+            value = "$\"Hello {Table({a: 5})} World!\"";
```

Comment:
Just to confirm, this line is a pure lexer test, since this is an invalid expression, correct?

Reply:
Yup. This test is just looking at how it's handling nested { within a {} section

Reply:
Got it. We could use something like `"$\"Hello {Table({a: 5}).a} World!\""` that would be a valid expression while still validating that. But if this is a lexer-only test we don't need to worry too much.

Reply:
Just a lexer test